common/streams/src/sources/source_ndjson.rs

```diff
@@ -123,11 +123,19 @@ where R: AsyncBufRead + Unpin + Send
             }
 
             let json: serde_json::Value = serde_json::from_reader(self.buffer.as_bytes())?;
-            // Deserialize each field.
 
-            for (name, deser) in names.iter().zip(packs.iter_mut()) {
+            for ((name, type_name), deser) in fields.iter().zip(packs.iter_mut()) {
                 let value = &json[name];
-                deser.de_json(value)?;
+                deser.de_json(value).map_err(|e| {
+                    ErrorCode::BadBytes(format!(
+                        "error at row {} column {}: type={}, value={:?}: {}",
+                        rows,
+                        name,
+                        type_name,
+                        value,
```

Comment:
If the value is larger, we need to truncate it to a MAX_LENGTH(1024)

Reply:
👌

Reply:
fixed in https://github.com/datafuselabs/databend/pull/4640